examples/benchmark.py

```diff
@@ -127,7 +127,6 @@
 
         for params in params_sweep:
             _, _, _, res = run(**params)
-            clear_cache()
```

Comment:
No, are you sure that not explicitly calling `clear_cache` still invokes `__del__` in the way we need it to clean up our objects correctly? If I recall correctly, removing the deep copy did not fix the ghost referencing from SymPy, so I am sceptical that this does what you think it does. Have you tried replacing the `clear_cache` call in `test_clear_cache` with an explicit GC to verify that `clear_cache` is obsolete now?